include/units.h

```diff
@@ -1912,7 +2051,8 @@ namespace units
 		template<class UnitTypeRhs, typename Ty, template<typename> class NlsRhs>
 		inline constexpr bool operator<(const unit<UnitTypeRhs, Ty, NlsRhs>& rhs) const noexcept
 		{
-			return (nls::m_value < units::convert<UnitTypeRhs, UnitType>(rhs.m_value));
+			using CommonUnit = std::common_type_t<unit, unit<UnitTypeRhs, Ty, NlsRhs>>;
+			return (CommonUnit(*this).m_value < CommonUnit(rhs).m_value);

```

Comment:
@nholthaus verify that `common_type` conversions have same of improved performance.